EasyMonads/Maybe/Core.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+
+namespace EasyMonads
+{
+   public static partial class Core
+   {
+      public static Maybe.None None => EasyMonads.Maybe.None.Default;
+
+      /// <summary>
+      /// Simple factory for `Maybe` type.
+      /// </summary>
+      /// <param name="value">non-null for `Some` or null for `None`</param>
+      /// <typeparam name="T">Type held by Maybe</typeparam>
+      /// <returns>Maybe</returns>
+      public static Maybe<T> Maybe<T>(T value) => new(value);
+
+      /// <summary>
+      /// Factory for `Maybe` type when value is expected to be non-null.
+      /// </summary>
+      /// <param name="value">non-null value</param>
+      /// <typeparam name="T">Type held by Maybe</typeparam>
+      /// <returns>Maybe</returns>
+      /// <exception cref="ArgumentNullException">Thrown if value is null</exception>
+      public static Maybe<T> Some<T>(T value)
+      {
+         if (value is null)
+         {
+            throw new ArgumentNullException(nameof(value));

```

Comment:
Granted `Some(null)` is invalid and should not proceed, still not sure I like the idea of throwing an exception in case that does occur.

What's the benefit to `var some = Some(5)` over `Maybe<int> some = 5`, which accomplishes the same thing, but also handles the implicit null-to-None case?

Reply:
* I want to guard against null when using `Some`. If you use this function, then you are providing some (non-null) value.
* And it allows the `Maybe<T>` ctor to still treat nulls as None if you want to use it directly 
* `Some(5)` looks clean (no need to explicitly specify the type)
* I don't want to `return Maybe<SomeClass>.None;`
  - I just want to `return None;`
  - and I really _really_ do not want to `return null;` when I want `None`

Reply:
Will probably get better once an action is taken on #16 .